src/nvim/pos.h

```diff
@@ -11,9 +14,9 @@ typedef int colnr_T;
 #define PRIdCOLNR "d"
 
 /// Maximal (invalid) line number
-enum { MAXLNUM = 0x7fffffff };
+#define MAXLNUM LONG_MAX
```

Comment:
This breaks buffer updates.

Tried changing this line to `const long MAXLNUM = LONG_MAX;` but it fails the static assertions where `SIZE_MAX >= MAXLNUM`.

Reply:
cc @phodge @KillTheMule

Reply:
So I started looking at this, but I'm confused on an early level. First, the assertions are justified because of the following casts, good we have them :)

But then, if `SIZE_MAX < MAXLNUM`, that means we're potentially having more lines than fit in a `size_t`. Is that a real usecase, can that happen? I'm on shaky ground here, but I always assumed `SIZE_MAX` is really the largest addressable thing. The vim issue doesn't really discuss things...

I'd have figured `#define MAXLUM SIZE_MAX` would be the way to go then. Since line numbers are `long`, things get complicated, but could we just take the smaller value of `LONG_MAX` and `SIZE_MAX`?

We've always left the question open on chunking updates, and I'm a tad unclear on how it could be done.

Reply:
> But then, if `SIZE_MAX < MAXLNUM`, that means we're potentially having more lines than fit in a `size_t`. Is that a real usecase, can that happen?

No, it can't happen.